Source/Dafny/Resolver.cs

```diff
@@ -2275,6 +2275,20 @@ public class ModuleBindings {
 
             extraMember.InheritVisibility(m, false);
             members.Add(extraName, extraMember);
+          } else if (m is Function f && f.ByMethodBody != null) {
+            var tok = f.ByMethodTok;
+            var resultVar = f.Result ?? new Formal(tok, "#result", f.ResultType, false, false, null);
+            var r = Expression.CreateIdentExpr(resultVar);
+            var receiver = f.IsStatic ? (Expression)new StaticReceiverExpr(tok, cl, true) : new ImplicitThisExpr(tok);
+            var fn = new FunctionCallExpr(tok, f.Name, receiver, tok, f.Formals.ConvertAll(Expression.CreateIdentExpr));
```

Comment:
The generated token would still need to map to some source location, and I think the "borrowed" source location is still the best one. Since nothing in the code needs to distinguish this token based on it being generated or not, I think the code is good as it is.